cmd/tidb-lightning-ctl/main.go

```diff
@@ -190,8 +190,55 @@ func checkpointRemove(ctx context.Context, cfg *config.Config, tableName string)
 		return errors.Trace(err)
 	}
 	defer cpdb.Close()
+	if err = cpdb.RemoveCheckpoint(ctx, tableName); err != nil {
+		return errors.Trace(err)
+	}
+
+	taskCp, err := cpdb.TaskCheckpoint(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	// a empty id means this task is not inited, we needn't further check metas.
+	if taskCp.TaskID == 0 {
+		return nil
+	}
+
+	// try to clean up table metas if exists
+	return cleanupMetas(ctx, cfg, tableName)
+}
+
+func cleanupMetas(ctx context.Context, cfg *config.Config, tableName string) error {
+	if tableName == "all" {
+		tableName = ""
+	}
+	// try to clean up table metas if exists
+	db, err := restore.DBFromConfig(cfg.TiDB)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	exist, err := common.TableExists(ctx, db, cfg.App.MetaSchemaName, restore.TableMetaTableName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if !exist {
```

Comment:
If you exit when `TableMetaTableName` doest not exist, you shall remove it at last.